File: caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

int stringToInt(string s);

int main(int argc, string argv[])
{
    int x;
    //check to make sure the user inputed one string
    if (argc == 2)
    {
        //check to make sure all inputed values are ints
        for (int i = 0, n = strlen(argv[1]); i < n; i++)
        {
            char c = argv[1][i];
            if (c < '0' || c > '9')
            {
                printf("Usage: ./caesar key \n");
                exit(0);
            }
        }
        //convert the string to an int
        x = stringToInt(argv[1]);
    }
    else
    {
        printf("Usage: ./caesar key \n");
        exit(0);
    }
    string m = get_string("plaintext: ");
    int l = strlen(m);
    char cipher [l];
    for (int i = 0; i < l; i++)
    {
        //check if char is uppercase
        if (isupper(m[i]))
        {
            //change the value of m[i] by x
            //use %26 to ensure that value remains in alphabet
            //if the shift is greater than 26, then the value wraps around to 'A'
            cipher[i] = (m[i] - 'A' + x) % 26 + 'A';
        }
        //check if char is lowercase
        else if (islower(m[i]))
        {
            //same as above but with lowercase
            cipher[i] = (m[i] - 'a' + x) % 26 + 'a';
        }
        //otherwise, do not shift.
        //Only alphabetical charachters rotate
        else
        {
            cipher[i] = m[i];
        }
    }
    //printf was not working properly for the string
    //this code prints the array of chars in proper format
    printf("ciphertext: ");
    for (int i = 0; i < l; i++)
    {
        printf("%c", cipher[i]);
    }
    printf("\n");
}

//convert the string to an int.  Made before I was aware of atoi.
int stringToInt(string s)
{
    int sum = 0;
    int ex = 1;
    //go backwar through the string, multiplying each value by 10
    //ie 123 = 3 + 20 + 100
    for (int i = strlen(s) - 1; i >= 0; i--)
    {
        int x = (int) s[i] - '0';
        sum += x * ex;
        ex *= 10;
    }
    return sum;
}

File: caesar.py
import sys
from cs50 import get_string

letters = 26


def main():
    # make sure user entered proper command line
    if len(sys.argv) != 2:
        print("Usage: python caesar.py k")
        sys.exit(1)
    # cast the user's input to an int
    x = int(sys.argv[1])

    # prompt user for message
    p = get_string("plaintext:  ")
    if not p:
        sys.exit(1)
    # create cipher to store new message
    cipher = ""
    # iterate through the message, letter by letter
    for c in p:
        # check if the char is upper, lower, or not alphabetical
        if(c.isupper()):
            # rotate by x, while using mod letters to ensure the letter remains within bounds of alphabet
            cipher += chr((ord(c) - ord('A') + x) % letters + ord('A'))
        elif(c.islower()):
            # same as above except for lowercase letters
            cipher += chr((ord(c) - ord('a') + x) % letters + ord('a'))
        else:
            # just add the non-alphabetical char to the message as is
            cipher += c
    # print the new message with
    print(f"ciphertext: {cipher}")


if __name__ == "__main__":
    main()
